packages/knip/src/WorkspaceWorker.ts

```diff
@@ -253,8 +253,16 @@ export class WorkspaceWorker {
         const patterns = this.getConfigurationFilePatterns(pluginName);
         const allConfigFilePaths = await _pureGlob({ patterns, cwd, gitignore: false });
 
+        const { packageJsonPath } = plugin;
         const configFilePaths = allConfigFilePaths.filter(
-          filePath => basename(filePath) !== 'package.json' || get(this.manifest, plugin.packageJsonPath ?? pluginName)
+          filePath =>
+            basename(filePath) !== 'package.json' ||
+            get(
+              this.manifest,
+              (Array.isArray(packageJsonPath)
+                ? packageJsonPath.find(path => get(this.manifest, path))
+                : packageJsonPath) ?? pluginName
+            )
```

Comment:
Just an idea, maybe it would be slightly simpler if we would accept a function instead of an array, something like this:

```ts
typeof packageJsonPath === 'function'
  ? packageJsonPath(this.manifest)
  : (get(this.manifest, plugin.packageJsonPath) ?? pluginName)
```

Not much better, but at least any (additional) complexity is moved to the plugin.

Reply:
> at least any (additional) complexity is moved to the plugin.

Certainly..., thank you...!
Fixed!
https://github.com/webpro/knip/pull/645/commits/58f13a794dae83fc8ced98bf57400985b7c3537a